### PR TITLE
github: Use new issue template feature

### PR DESCRIPTION
**Pull request by emilengler**
This helps us to avoid getting feature requests as well as security
bugs reported into the issue tracker.

### CODE REVIEW

**Comment by mback2k** [comment]
Instead of providing templates with these warnings it is also possible to directly link to the corresponding pages, eg.:
https://github.com/home-assistant/core/blob/dev/.github/ISSUE_TEMPLATE/config.yml

**Comment by emilengler** [comment]
> Instead of providing templates with these warnings it is also possible to directly link to the corresponding pages, eg.:
> 
> https://github.com/home-assistant/core/blob/dev/.github/ISSUE_TEMPLATE/config.yml

Thanks, TIL

File: .github/ISSUE_TEMPLATE/security-vulnerability.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+about: 'View our security policy here: https://curl.haxx.se/docs/security.html'
+title: 'View our security policy here: https://curl.haxx.se/docs/security.html'
+labels: ''
+assignees: ''
+
+---
+
+View our security policy here: https://curl.haxx.se/docs/security.html
+
+DO NOT REPORT THE ISSUE PUBLICLY ON GITHUB
```
Comment (bagder) [inline .github/ISSUE_TEMPLATE/security-vulnerability.md:12]: Wouldn't it be better if this instead linked directly to the hackerone page where user would submit the security issue? https://hackerone.com/curl

**Comment by emilengler** [comment]
Fixed. Now security redirects to hackerone and feature request to the ML.
I think this is ready for merge

**Comment by emilengler** [comment]
I think it's fine to leave the comments in the bug_report.md file though

**Comment by emilengler** [comment]
For the case someone wants to test it: https://github.com/emilengler/curl-issue-test/issues/new/choose

File: .github/ISSUE_TEMPLATE/config.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+blank_issues_enabled: false
+contact_links:
+  - name: Request a feature
+    url: http://cool.haxx.se/cgi-bin/mailman/listinfo/curl-library
+    about: This is only for bugs! Feature requests should go to the ML
```
Comment (mback2k) [inline .github/ISSUE_TEMPLATE/config.yml:5]: I would spell it out as mailing list instead of ML so that newcomers understand it better.

Comment (mback2k) [reply .github/ISSUE_TEMPLATE/config.yml:5]: Maybe the link should also be this page instead: https://curl.haxx.se/mail/

**Comment by emilengler** [comment]
@mback2k Adjustments applied

**Comment by bagder** [comment]
That test link isn't showing the latest update, does it?

**Comment by emilengler** [comment]
> That test link isn't showing the latest update, does it?

Nope but it's only a text change. I can't update it from here unfortunately. I am on the ride

**Comment by bagder** [comment]
Fourth entry "Question" and instruct that questions should be taken to a dedicated mailing list.

A possible fifth entry could be "Commercial support" and saying that if the user is interested in hiring/paying someone for support, we provide alternatives on https://curl.haxx.se/support.html

**Comment by emilengler** [comment]
I implemented @bagder's proposals

**Comment by bagder** [comment]
Awesome, can you refresh your example URL so that we can see how this looks "live" ?

**Comment by emilengler** [comment]
> Awesome, can you refresh your example URL so that we can see how this looks "live" ?

Yes I will do this in the afternoon!

**Comment by emilengler** [comment]
@bagder Updated

**Comment by emilengler** [comment]
I think this is ready for merge now, the demo is also updated

**Review by bagder — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/config.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+blank_issues_enabled: false
+contact_links:
+  - name: Feature request
+    url: https://curl.haxx.se/mail/
+    about: To propose new features or enhancements, please bring that discussion to a suitable curl mailing list.
+  - name: Security vulnerability
+    url: https://hackerone.com/curl
```
Comment (mback2k) [inline .github/ISSUE_TEMPLATE/config.yml:7]: Would it make sense to link to https://curl.haxx.se/dev/secprocess.html instead?

**Review by mback2k — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/config.yml (hunk 3 above)
Comment (bagder) [reply .github/ISSUE_TEMPLATE/config.yml:7]: I don't think so. The secprocess page is more about describing how we approach it, if someone wants to report an issue I figured the hackerone page is more direct.

**Comment by jay** [comment]
There's some talk earlier about a demo, how do I access that?

**Comment by bagder** [comment]
@emilengler's demo is at https://github.com/emilengler/curl-issue-test/issues/new/choose

**Review by jay — APPROVED** [review]

**Comment by bagder** [comment]
Thanks!

**Comment by bagder** [comment]
Hm, annoyingly enough it seems GitHub adds an extra "Report a security vulnerability" by itself:
Screenshot_2020-09-17 curl curl

I guess we need to remove our own version of it then ....

**Pull request closed without merging**